### ISSUE TITLE
Possible cyclic (circular) dependencies causing issues with Webpack

### ISSUE THREAD

**Issue by rquast** [substance/substance#1092]
I've been using webpack to bundle substance. It worked with beta 6 until I tried beta 7.

I managed to get it to work again by changing the following:

```
import { getSelectionRect, getRelativeMouseBounds, platform } from '../util'
import { DefaultDOMElement } from '../dom'
import Component from '../ui/Component'

class AbstractScrollPane extends Component {
```
I changed import { Component } form '../ui' to skip the re-export.

I've tried checking for cyclic dependencies with madge, but I think it crashes each time it hits one.

**Comment by obuchtala** [comment]
Somehow `Rollup` is able to deal with the code. At the beginning we had to resolve real cycles such as  `Selection.js` having a factory method depending on `PropertySelection.js` which apparently extended `Selection` exported by `Selection.js`.
`webpack` seems to be choking with the `index.js` files.
Have you tried to separate webpack and Substance bundling? I.e. using the single file bundle?

**Comment by rquast** [comment]
Yes, I tried to use the single file bundle, but for some reason or other, it's very slow for webpack to process. I might try forking the project and making a webpack build to see if I can recreate the problem, at least to test new builds.

Other than that small thing, everything works perfectly.

**Comment by obuchtala** [comment]
Ok, it's slow, because it really needs to analyse the whole library, and can not drop things on the fly.

**Comment by obuchtala** [comment]
Maybe it is an option for you to load `substance.js` via an extra `<script>` and tell webpack to treat it as external located at `window.substance`?

Edit: this is what we do in Texture, too

**Comment by rquast** [comment]
That's probably a good idea too. I wasn't really keen on splitting any libraries from the bundle, but the editor is a good exception because of the size.
